static/app/views/organizationContextContainer.tsx

```diff
@@ -16,329 +14,213 @@ import {ORGANIZATION_FETCH_ERROR_TYPES} from 'sentry/constants';
+  useEffect(() => {
+    // Nothing to do if we already have the organization loaded
+    if (organization && organization.slug === orgSlug) {
       return;
     }
 
-    metric.mark({name: 'organization-details-fetch-start'});
-    fetchOrganizationDetails(
-      this.props.api,
-      orgSlug,
-      !isOrgChanging(this.props), // if true, will preserve a lightweight org that was fetched,
-      isInitialFetch
-    );
-  }
+    handleLoad();
+  }, [orgSlug, organization, handleLoad]);
 
-  loadOrganization(orgData: State) {
-    const {organization, error} = orgData;
+  // Take a measurement for when organization details are done loading and the
+  // new state is applied
+  useEffect(
+    () => {
+      if (organization === null) {
+        return;
+      }
 
-    if (organization && !error) {
-      // Configure scope to have organization tag
-      Sentry.configureScope(scope => {
-        // XXX(dcramer): this is duplicated in sdk.py on the backend
-        scope.setTag('organization', organization.id);
-        scope.setTag('organization.slug', organization.slug);
-        scope.setContext('organization', {id: organization.id, slug: organization.slug});
+      metric.measure({
+        name: 'app.component.perf',
+        start: 'organization-details-fetch-start',
+        data: {
+          name: 'org-details',
+          route: getRouteStringFromRoutes(routes),
+          organization_id: parseInt(organization.id, 10),
+        },
       });
-    } else if (error) {
-      // If user is superuser, open sudo window
-      const user = ConfigStore.get('user');
-      if (!user || !user.isSuperuser || error.status !== 403) {
-        // This `catch` can swallow up errors in development (and tests)
-        // So let's log them. This may create some noise, especially the test case where
-        // we specifically test this branch
-        console.error(error); // eslint-disable-line no-console
-      } else {
-        openSudo({
-          retryRequest: () => Promise.resolve(this.fetchData()),
-          isSuperuser: true,
-          needsReload: true,
-        });
-      }
+    },
+    // Ignore the `routes` dependency for the metrics measurement
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [organization]
+  );
+
+  // Configure sentry SDK scope to have organization tag
+  useEffect(() => {
+    if (organization === null || error) {
+      return;
     }
 
-    this.setState({...orgData}, () => {
-      // Take a measurement for when organization details are done loading and the new state is applied
-      if (organization) {
-        metric.measure({
-          name: 'app.component.perf',
-          start: 'organization-details-fetch-start',
-          data: {
-            name: 'org-details',
-            route: getRouteStringFromRoutes(this.props.routes),
-            organization_id: parseInt(organization.id, 10),
-          },
-        });
-      }
+    Sentry.configureScope(scope => {
```

Comment:
Since this runs in a useEffect, it might happen that the children components triggering sentry related events won't have these properties applied as the effect will run after the component (for example if a child throws a sync error crashing the page, this effect will never run)

A solution would be to inline it like
```js
const initializedSentryRef = useRef(false)

if(!initializedSentryRef.current && organization){
  Sentry.configureScope...
  initializedSentryRef.current = true
}
```